pkg/destroy/aws/aws.go

```diff
@@ -785,22 +812,29 @@ func terminateEC2InstancesByTags(ec2Client *ec2.EC2, iamClient *iam.IAM, filters
 // Profiles.
 //
 // This code is a place to find specific objects like this which might be dangling.
-func (o *ClusterUninstaller) deleteUntaggedResources(awsSession *session.Session) error {
+func (o *ClusterUninstaller) deleteUntaggedResources(ctx context.Context, awsSession *session.Session) (resources destroyedResources, returnErr error) {
 	iamClient := iam.New(awsSession)
-	masterProfile := fmt.Sprintf("%s-master-profile", o.ClusterID)
-	if err := deleteIAMInstanceProfileByName(iamClient, &masterProfile, o.Logger); err != nil {
-		return err
-	}
-	workerProfile := fmt.Sprintf("%s-worker-profile", o.ClusterID)
-	if err := deleteIAMInstanceProfileByName(iamClient, &workerProfile, o.Logger); err != nil {
-		return err
+	resources = destroyedResources{}
+	for _, profileType := range []string{"master", "worker"} {
+		profile := fmt.Sprintf("%s-%s-profile", o.ClusterID, profileType)
+		response, err := iamClient.GetInstanceProfileWithContext(ctx, &iam.GetInstanceProfileInput{InstanceProfileName: &profile})
+		if err != nil {
+			if err.(awserr.Error).Code() == iam.ErrCodeNoSuchEntityException {
```

Comment:
:+1:

Reply:
Does this need a get preflight? Looks like we catch and ignore this from the delete itself in `deleteIAMInstanceProfileByName` since the function landed in #1268.

Reply:
The preflight is so that I can get the ARN to record whether it was deleted successfully.

Reply:
Can we update `deleteIAMInstanceProfileByName` to return the deleted ARN?

Reply:
We can. But that function does not know the ARN either, so it would have to look it up. The other caller of the function already knows the ARN, so it would be a wasted request for that caller.

Reply:
It looks like I can get the deduce the ARN without looking up the instance profile. Let me change it to do that instead.

Reply:
Actually, it is not that straight-forward. The account ID is not known in that function.